pennylane/ops/op_math/sum.py

```diff
@@ -0,0 +1,286 @@
+
+
+def _sum(mats_gen, dtype=None, cast_like=None):
+    r"""Private method to compute the sum of matrices.
+
+    Args:
+        mats_gen (Generator): a python generator which produces the matricies which
+            will be summed together.
+
+    Keyword Args:
+        dtype (str): a string representing the data type of the entries in the result.
+        cast_like (Tensor): a tensor with the desired data type in its entries.
+
+    Returns:
+        res (Tensor): the tensor which is the sum of the matrices obtained from mats_gen.
+    """
+    # Note this method is currently inefficient (improve addition by looking at wire subgroups)
+    res = reduce(math.add, mats_gen)
+
+    if dtype is not None:
+        res = math.cast(res, dtype)
+    if cast_like is not None:
+        res = math.cast_like(res, cast_like)
+
+    return res
+
+
+class Sum(Operator):
+    r"""Symbolic operator representing the sum of operators.
+
+    Args:
+        summands (tuple[~.operation.Operators]): a tuple of operators which will be summed together.
+
+    Keyword Args:
+        do_queue (bool): determines if the sum operator will be queued (currently not supported).
+            Default is True.
+        id (str or None): id for the sum operator. Default is None.
+
+    Returns:
+        ~ops.op_math.Sum: the operator representing the sum of summands.
+
+    ..seealso:: :class:`~.ops.op_math.Sum`
+
+    **Example**
+
+    >>> summed_op = op_sum(qml.PauliX(0), qml.PauliZ(0))
+    >>> summed_op
+    PauliX(wires=[0]) + PauliZ(wires=[0])
+    >>> qml.matrix(summed_op)
+    array([[ 1,  1],
+           [ 1, -1]])
+    >>> summed_op.terms()
+    ([1.0, 1.0], (PauliX(wires=[0]), PauliZ(wires=[0])))
+
+    .. details::
+        :title: Usage Details
+
+        We can combine parameterized operators, and support sums between operators acting on
+        different wires.
+
+        >>> summed_op = op_sum(qml.RZ(1.23, wires=0), qml.Identity(wires=1))
+        >>> summed_op.matrix()
+        array([[1.81677345-0.57695852j, 0.        +0.j        ,
+                0.        +0.j        , 0.        +0.j        ],
+               [0.        +0.j        , 1.81677345-0.57695852j,
+                0.        +0.j        , 0.        +0.j        ],
+               [0.        +0.j        , 0.        +0.j        ,
+                1.81677345+0.57695852j, 0.        +0.j        ],
+               [0.        +0.j        , 0.        +0.j        ,
+                0.        +0.j        , 1.81677345+0.57695852j]])
+    """
+
+    _eigs = {}  # cache eigen vectors and values like in qml.Hermitian
+
+    def __init__(self, *summands, do_queue=True, id=None):
+        """Initialize a Symbolic Operator class corresponding to the Sum of operations."""
+        self._name = "Sum"
+        self._id = id
+        self.queue_idx = None
+
+        if len(summands) < 2:
+            raise ValueError(f"Require at least two operators to sum; got {len(summands)}")
```

Comment:
Make sure to test initialization logic. Looks like you also don't have tests for `name`, `id`, and `queue_idx`.  `parameters` and `data` could also use unit testing.